packages/fxa-content-server/app/scripts/views/mixins/email-autocomplete-domains-mixin.js

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * A mixin to autocomplete common domains from `datalist option` `value`s.
+ */
+import Template from 'templates/partial/email-autocomplete-domains.mustache';
+import UserAgentMixin from '../../lib/user-agent-mixin';
+
+export const EMAIL_SELECTOR = 'input[type=email]';
+export const DATALIST_OPTIONS_SELECTOR = '#autocomplete-domain option';
+export const FOCUS_HACK_SELECTOR = '#focus-hack';
+
+export const DOMAINS = [
+  'gmail.com',
+  'outlook.com',
+  'hotmail.com',
+  'yahoo.com',
+  'qq.com',
+  'web.de',
+  'aol.com',
+  'mail.ru',
+  'icloud.com',
+  'gmx.de',
+  't-online.de',
+  'orange.fr',
+  'yandex.ru',
+  'yahoo.fr',
+  'live.com',
+  '163.com',
+  'msn.com',
+  'comcast.net',
+  'hotmail.co.uk',
+  'hotmail.fr',
+];
+
+export default {
+  events: {
+    [`keyup ${EMAIL_SELECTOR}`]: '_toggleDomainAutocomplete',
+  },
+  dependsOn: [UserAgentMixin],
+
+  setInitialContext(context) {
+    const isAndroid = this.getUserAgent().isAndroid();
+    const isChrome = this.getUserAgent().isChrome();
+
+    const unsafeEmailAutocompleteDomainsHTML = this.renderTemplate(Template, {
+      options: DOMAINS,
+    });
+    context.set({
+      unsafeEmailAutocompleteDomainsHTML,
+      isAndroid,
+      isChrome,
+    });
+  },
+
+  afterRender() {
+    this.input = this.$(EMAIL_SELECTOR);
+    this.datalistOptions = this.$(DATALIST_OPTIONS_SELECTOR);
+    this.focusHack = this.$(FOCUS_HACK_SELECTOR);
+    this.previousUsername = null;
+  },
+
+  /**
+   * A catch-most-browsers hack when datalist options are modified.
+   * Chrome otherwise shows all options when '@' is typed,
+   * Safari immediately goes to the dropdown preventing users
+   * from typing further, and FF will not update the dropdown on `keyup`
+   * and has a known bug with dynamic datalists
+   * https://bugzilla.mozilla.org/show_bug.cgi?id=1474137
+   */
+  _focusHack() {
+    this.focusHack.focus();
+    this.input.focus();
+  },
+
+  _toggleDomainAutocomplete() {
+    const inputValue = this.input.val();
+    const username = inputValue.substr(0, inputValue.indexOf('@'));
+
+    if (
+      inputValue.length >= 2 &&
+      inputValue.includes('@') &&
+      ((this.previousUsername !== null && username !== this.previousUsername) ||
+        !this.input.attr('list'))
+    ) {
+      const { isAndroid, isChrome } = this.getContext();
+      // Chrome and Android browsers show all options when '@' is typed.
+      // Non-Android Chrome is covered in `_focusHack` but if the device is
+      // Android we must wait until there is one additional character.
+      if (
+        !isAndroid ||
+        (isAndroid && inputValue.charAt(inputValue.length - 1) !== '@')
```

Comment:
You don't really need the `isAndroid &&` here, because if the first condition isn't met, `isAndroid` is guaranteed to be truthy. It also looks like `inputValue.charAt(inputValue.length - 1) !== '@'` could be simplified by using [String.prototype.endsWith](https://developer.mozilla.org/docs/Web/JavaScript/Reference/Global_Objects/String/endsWith).